grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -340,9 +338,11 @@ class IdentifierService {
             }
         }
 
-        // If product type is not deafult and has sequential identifier format or code,
-        // then generate sequentional, otherwise random
-        if (productType?.productIdentifierFormat?.contains("0") || productType?.code) {
+        // If product type is not default
+        // and has sequential identifier format containing "0" character which indicates the sequential part
+        // or identifier format is not specified when generatorTypeCode is SEQUENCE but has a code
+        // then generate sequential, otherwise random
+        if (productType?.productIdentifierFormat?.contains("0") || (!productType?.productIdentifierFormat && productType?.code && generatorTypeCode == IdentifierGeneratorTypeCode.SEQUENCE)) {
```

Comment:
this long if statement bothers me a bit and probably would be better to put it in a named variable, but I am unsure of the naming for it.
Maybe (probably not) `Boolean isSequentialWithCodeAndNoPIF`
If you have any suggestions let me know or we can leave it as is.

Reply:
I agree but can't think of a good name either. Maybe it's less about the specific logic and more about what you plan to do if it's true.
```
Boolean requireSequentialIdentifier = ...
```

Reply:
@alannadolny since Dariusz is out of the office today, could you make this small improvement?

Reply:
@alannadolny since Dariusz is out of the office today, could you make this small improvement?